extension/icu/icu-datepart.cpp

```diff
@@ -355,6 +363,29 @@ struct ICUDatePart : public ICUDateFunc {
 		result.Verify(count);
 	}
 
+	template <typename INPUT_TYPE>
+	static void NameFunction(DataChunk &args, ExpressionState &state, Vector &result) {
```

Comment:
I originally tried to use `UnaryTimestampFunction<INPUT_TYPE, string_t>` but it didn't seem to like how the last branch in the function passed to `ExecuteWithNulls` would end up doing `string_t(0)` - this function is essentially the same but does a no-arg `string_t()` in its place. Perhaps there's an easy way to unify these?

Reply:
The result of the function is ignored, so just change it to `return RESULT_TYPE();`

Reply:
Thanks for the quick response! I've taken a crack at refactoring to use `UnaryTimestampFunction`.

For a little more background: We ([Hex](https://hex.tech/)) integrate DuckDB into our product. When we upgraded to 0.5.0 last year it fixed an issue with Pandas tz-aware columns being preserved, which was great, but presented as a somewhat breaking change from the perspective of our users who were used to all timestamp columns ending up as naive timestamps. It turned out the change would break quite a few existing user projects so we made the call to apply a couple of small patches to preserve compatibility for the most common breakages: the ability to cast to `date` (even though it couldn't respect tz) and these `dayname`/`monthname` functions.
I noticed that you recently added a [proper date cast](https://github.com/duckdb/duckdb/pull/7150), and realized that if we could upstream these functions it would mean we could drop our fork on the next upgrade.